docs/code-guide.md

```diff
@@ -0,0 +1,133 @@
+# Coding Guidelines
+
+This document defines the coding standards for the OCM UI project. All code contributions must follow these guidelines to ensure consistency, maintainability, and code quality.
+
+These guidelines serve to:
+- Establish shared conventions for code structure and patterns
+- Improve code reviews by setting clear expectations
+- Maintain a consistent, maintainable codebase
+
+Use these guidelines during development and code reviews. When in doubt or if a scenario isn't covered, ask the team for help.
+
+## Component Structure
+
+Components should be organized by features. Each feature is represented by a container component in charge of handling logic and rendering other presentational components. The main responsibilities of presentational components are to display data received via props and handle user interactions.
+
+Components should be properly broken down into UI elements responsible for a single goal. 
+
+Example folder structure for a hypothetical `UsersList` component:
+
+```
+UsersList/
+├── UsersList.tsx           # Container component
+├── UsersList.test.tsx      
+├── UsersList.stories.tsx      
+├── useUsersList.ts         # Feature-specific hooks
+├── components/             # Presentational components (only imported by container)
+├── └── UsersTable.tsx      # Only create a folder if you have many
+├── └── UsersToolbar.tsx
+├── types.ts                # Only if needed      
+└── utils/                  # Feature-specific utilities
+```
+
+## Coding Patterns
+
+Use the following patterns when designing components:
+- UI components should be a thin wrapper around data, they should handle local state only when necessary.
+- Try to flatten the UI state into a basic calculation, deriving data from props
```

Comment:
'flatten the UI state into a basic calculation' sounds kinda vague; i didn't fully understand that.

can you rephrase for clarity, or elaborate?

Reply:
'flatten the UI state into a basic calculation' still sounds vague, and tbh, so is 'eventually do basic manipulations/calculations with constants'.  it'd be nice to elaborate more or add a concrete example in that bullet.

is this basically about the 'pull state up' principle?  can we maybe write that?

Reply:
Ok I've combined this point and the previous one. They are related and doing so could help to clarify the big picture.